src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -200,6 +214,100 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+                        where outSymbol.Equals(symbol)
+                        select id;
+
+            var references = query.ToImmutableArray<SyntaxNode>();
+
+            var root = semanticModel.SyntaxTree.GetCompilationUnitRoot(cancellationToken);
+
+            // Ensure we can track the references and the local variable as we make edits
+            // to the tree.
+            var rootWithTrackedNodes = root.TrackNodes(references.Concat(localDeclarator).Concat(enclosingBlock));
+
+            // Now, take the local variable and remove it's initializer.  Then go to all
+            // the locations where we read from it.  If they're definitely assigned, then
+            // that means the out-var did it's work and assigned the variable across all
+            // paths. If it's not definitely assigned, then we can't inline this variable.
+            var currentLocalDeclarator = rootWithTrackedNodes.GetCurrentNode(localDeclarator);
+            var rootWithoutInitializer = rootWithTrackedNodes.ReplaceNode(
+                currentLocalDeclarator,
+                currentLocalDeclarator.WithInitializer(null));
+
+            // Fork the compilation so we can do this analysis.
+            var newCompilation = semanticModel.Compilation.ReplaceSyntaxTree(
+                root.SyntaxTree, rootWithoutInitializer.SyntaxTree);
+            var newSemanticModel = newCompilation.GetSemanticModel(rootWithoutInitializer.SyntaxTree);
+
+            // NOTE: there is no current compiler API to determine if a variable is definitely
+            // assigned or not.  So, for now, we just get diagnostics for this block and see if
+            // we get any definite assigment errors where we have a reference to the symbol. If
+            // so, then we don't offer the fix.
+
+            var currentBlock = rootWithoutInitializer.GetCurrentNode(enclosingBlock);
+            var diagnostics = newSemanticModel.GetDiagnostics(currentBlock.Span, cancellationToken);
+
+            var diagnosticSpans = diagnostics.Where(d => d.Id == CS0165)
+                                                .Select(d => d.Location.SourceSpan)
+                                                .Distinct();
+
+            var newReferenceSpans = rootWithoutInitializer.GetCurrentNodes<SyntaxNode>(references)
+                                                            .Select(n => n.Span)
+                                                            .Distinct();
+
+            return diagnosticSpans.Intersect(newReferenceSpans).Any();
+
+            // var annotation = new SyntaxAnnotation();
+            // var declaratorWithoutInitializer = rootWithoutInitializer.GetAnnotatedNodes(annotation)
+            //                                                         .Single();
+            // var currentLocalSymbol = (ILocalSymbol)newSemanticModel.GetDeclaredSymbol(declaratorWithoutInitializer);
+#if false
```

Comment:
Code that i would like to write, but cannot until https://github.com/dotnet/roslyn/issues/15430 completes.